### PR TITLE
Work in progress, add llm based scanning

### PR DESCRIPTION
**Pull request by SamSaffron**
This PR adds support for:

1. first_post_only trigger and first_topic_only trigger for post_created_edited this allows up only to apply a rule to the first post / topic a user makes

2. run_in_background support, which allows a scriptable to be fired in a background job and not block

Actual llm scanning support is still pending need the basics first.

### CODE REVIEW

File: app/jobs/regular/discourse_automation_trigger.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Jobs
+  class DiscourseAutomationTrigger < ::Jobs::Base
+    def execute(args)
+      automation = DiscourseAutomation::Automation.find_by(id: args[:automation_id])
```
Comment (jjaffeux) [inline app/jobs/regular/discourse_automation_trigger.rb:6]: you probably want to find by `enabled: true` too, might want to make it a default scope, but I have been bitten too many times in my life by default scopes...

File: lib/discourse_automation/scripts/llm_triage.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+DiscourseAutomation::Scriptable.add(DiscourseAutomation::Scriptable::LLM_TRIAGE) do
+  version 1
+
+  placeholder :creator_username
+
+  triggerables %i[post_created_edited]
+
+  script { |context, fields, automation| }
+end
```
Comment (jjaffeux) [inline lib/discourse_automation/scripts/llm_triage.rb:13]: Maybe it's just here temporary but you can add it to discourse-ai plugin like this:

```ruby
  if defined?(DiscourseAutomation)
    add_automation_scriptable("llm_triage") do
      placeholder :creator_username

      triggerables %i[post_created_edited]

      script do |context, fields, automation|
   
      end
    end
  end
```

Then you will need locales like this:

```yml
en:
  js:
    discourse_automation:
      scriptables:
        llm_triage:
           ...
```

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -55,6 +55,9 @@ en:
         title: After post cook
         description: When a post content is cooked the automation will be triggered
     scriptables:
+      llm_triage:
```
Comment (tgxworld) [inline config/locales/server.en.yml:58]: Just a note here, I’m not sure if AI stuff should be leaking into this plugin? Can this plugin perhaps expose APIs that can be registered from discourse-ai instead?

Comment (jjaffeux) [reply config/locales/server.en.yml:58]: look my comment above allan, there's already one: https://github.com/discourse/discourse-automation/pull/217#discussion_r1340962110

Comment (tgxworld) [reply config/locales/server.en.yml:58]: Oops sorry didn’t see that one.

Comment (jjaffeux) [reply config/locales/server.en.yml:58]: no worries

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**